framework/auth/utils.py

```diff
@@ -1,4 +1,46 @@
+import re
 from nameparser.parser import HumanName
+from modularodm.exceptions import ValidationError
+
+
+USER_REGEX = re.compile(
```

Comment:
Did you get this code from a 3rd-party source? If so, be sure to provide proper attribution.